backend/main.py

```diff
@@ -832,12 +833,13 @@ def filter_pipeline(payload, user):
             else:
                 pass
 
-    if "pipeline" not in app.state.MODELS[model_id]:
-        if "title" in payload:
-            del payload["title"]
-
-        if "task" in payload:
-            del payload["task"]
+    keep_extras = (
+        "pipeline" in app.state.MODELS[model_id] or "pipe" in app.state.MODELS[model_id]
```

Comment:
Only pipelines should have additional fields and `pipe` function should not have any extras.

Reply:
Is there any other way to know whether a function is being called for title/task/tool use? Currently I'm just checking for keywords in the prompt but I feel like that's not particularly robust.

Reply:
I'll try to think of something for this

Reply:
OK thanks! I've reverted this for now.